src/downgrade

```diff
@@ -536,7 +564,33 @@ cli() {
 
   # Proceed with rest of workflow
   main "${terms[@]}"
-  pacman -U "${pacman_args[@]}" "${to_install[@]}"
+  if [[ $DOWNGRADE_FROM_GIT ]]; then
+    for item in "${to_install[@]}"; do
+      if [[ ${item:0:1} != '/' ]] && [[ ${item:0:5} != "https" ]];then
+        commit_hash=${item%%-*}
+        path="$(echo "$item" | sed 's/[a-z0-9]*-\(.*\)-[0-9.]*.-any.pkg.tar.gz/\1/')"
+        local new_list=()
+        for i in "${to_install[@]}";do
+          if [[ "$i" != "$item" ]]; then
+            new_list+=("$i")
+          fi
+        done
+        to_install=("${new_list[@]}")
+
+        pushd "$path" >/dev/null
+        su -P "$(stat -c"%U" PKGBUILD)" -c "git checkout $commit_hash &>/dev/null && makepkg -si"
```

Comment:
I get what you're doing now, but I wonder if it would be simpler to _not_ add `-i` here.

If  `makepkg` built (but didn't install) a package with the same name that's already in the `to_install` array, you can just leave that array alone and have `pacman -U` handle it later like usual.

Would that work?